baybe/constraints/continuous.py

```diff
@@ -115,27 +130,51 @@ def to_botorch(
         Args:
             parameters: The parameter objects of the continuous space.
             idx_offset: Offset to the provided parameter indices.
+            batch_size: The batch size used for the recommendation. Necessary
+                for interpoint constraints as indices need to be adjusted.
+                Ignored by all other constraints.
+
+        Raises:
+            ValueError: When the constraint is an interpoint constraint but
+                no batch size is given or when providing a batch size while the
+                constraint is an intrapoint constraint.
 
         Returns:
             The tuple required by botorch.
         """
+        if (batch_size is not None) ^ (self.is_interpoint):
+            raise ValueError(
+                "A batch size must be set if and only if the constraint is "
+                "an interpoint constraint."
+            )
+
         import torch
 
         from baybe.utils.torch import DTypeFloatTorch
 
-        param_names = [p.name for p in parameters]
-        param_indices = [
-            param_names.index(p) + idx_offset
-            for p in self.parameters
-            if p in param_names
-        ]
+        coefficients = torch.tensor(self.coefficients, dtype=DTypeFloatTorch)
+
+        # Locate where the parameters referenced by the constraint will later be found
+        # in the input tensor provided to BoTorch
+        # NOTE: We assume here that all referenced parameters actually exist in the
+        #       provided ``parameters`` list, which is pre-validated when creating the
+        #       search space. If this assumption is violated for whatever reason, an
+        #       exception would still be thrown by the lookup attempt below.
+        names = [p.name for p in parameters]
+        idxs = torch.tensor([names.index(p) + idx_offset for p in self.parameters])
+
+        if self.is_interpoint:
+            # For interpoint constraints, the coefficients and indices need to be
+            # adjusted to account for the batch size:
+            # https://github.com/pytorch/botorch/blob/1518b304f47f5cdbaf9c175e808c90b3a0a6b86d/botorch/optim/optimize.py#L609 # noqa: E501
+            assert batch_size is not None
+            coefficients = coefficients.repeat_interleave(batch_size)
```

Comment:
@AVHopp: There is something going on here which I don't understand. Give me a ping via teams once you have time to discuss

Reply:
As discussed via teams, this is probably due to a bug in the new polytope sampling code, which somehow seems to implicitly rely on the hard-coded ordering of this method (which is ideally to be avoided – perhaps the problem becomes obsolete once [this issue](https://github.com/emdgroup/baybe/pull/625#discussion_r2398023959) has been fixed)